Pod/Classes/GraphQL/ARGraphQLQueryCache.m

```diff
@@ -88,6 +140,14 @@ - (instancetype)init;
     return [validUntil compare:[NSDate date]] == NSOrderedDescending;
 }
 
+static NSDate *
```

Comment:
@alloy is there an advantage to using static methods vs instance ones?

Reply:
These methods are ultimately getting called from `+load`, which is a class method that's invoked when the Objective-C runtime is loaded. There is no instance to call these on (or, if you prefer, the class itself _is_ the instance).

Reply:
Yeah besides it not really being possible in this case, as @ashfurrow said, there’s also the issue that these functions don’t really rely on any instance state (they are stateless). In those case I prefer to move a method out of a class into a function.

Finally, static functions will likely be inlined by the compiler whereas methods are dynamic and will never be inlined. Not that that optimization is really needed here, just one extra pro for static functions when all else is equal.

Reply:
Got it, so it's essentially another way to implement the singleton pattern. Thanks @ashfurrow

Reply:
Yes exactly – you can think of Objective-C classes / class methods as singleton objects. Swift takes this a step further: every instance method is _literally_ the return value of an identically-named class method that takes a `self` argument and returns a curried function (the instance method).

Reply:
This wa a good thread, thanks for asking @l2succes